src/goto-harness/recursive_initialization.cpp

```diff
@@ -1008,6 +1008,7 @@ code_blockt recursive_initializationt::build_function_pointer_constructor(
 
   const auto function_pointer_selector =
     get_fresh_local_symexpr("function_pointer_selector");
+  body.add(code_declt{function_pointer_selector});
```

Comment:
:question: Is there a test for this bug? That is, a test that passes if and only if this line is present

Reply:
The existing function pointer tests in goto harness wouldn't really work without this.